app/uk/gov/hmrc/eoricommoncomponent/frontend/connector/VatControlListConnector.scala

```diff
@@ -50,24 +49,24 @@ class VatControlListConnector @Inject() (httpClient: HttpClientV2, appConfig: Ap
       logger.debug(s"vat-known-facts-control-list successful. url: $url")
       // $COVERAGE-ON
       response.status match {
-        case OK        => Right(response.json.as[VatControlListResponse])
+        case OK        => handleResponse[VatControlListResponse](response)
         case NOT_FOUND =>
           // $COVERAGE-OFF$Loggers
           logger.warn(
             s"VatControlList failed. url: $url. Reason: The back end has indicated that vat known facts cannot be returned."
           )
           // $COVERAGE-ON
-          Left(NotFoundResponse)
+          Left(ResponseError(NOT_FOUND, response.body))
         case BAD_REQUEST =>
           // $COVERAGE-OFF$Loggers
           logger.warn(s"VatControlList failed. url: $url. Reason: Request has not passed validation. Invalid vrn.")
           // $COVERAGE-ON
-          Left(InvalidResponse)
+          Left(ResponseError(BAD_REQUEST, response.body))
         case SERVICE_UNAVAILABLE =>
           // $COVERAGE-OFF$Loggers
           logger.warn(s"VatControlList failed. url: $url. Reason: Dependent systems are currently not responding")
           // $COVERAGE-ON
-          Left(ServiceUnavailableResponse)
+          Left(ResponseError(SERVICE_UNAVAILABLE, response.body))
         case _ =>
           val error = "Incorrect VAT Known facts response"
           // $COVERAGE-OFF$Loggers
```

Comment:
For this final case can we change this to be a left as well instead of throwing an exception?

Reply:
👍